cobigen/cobigen-tsplugin/src/main/java/com/devonfw/cobigen/tsplugin/inputreader/TypeScriptInputReader.java

```diff
@@ -0,0 +1,289 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.devonfw.cobigen.api.constants.ExternalProcessConstants;
+import com.devonfw.cobigen.api.exception.InputReaderException;
+import com.devonfw.cobigen.api.extension.InputReader;
+import com.devonfw.cobigen.impl.exceptions.ConnectionExceptionHandler;
+import com.devonfw.cobigen.impl.externalprocess.ExternalProcessHandler;
+import com.devonfw.cobigen.tsplugin.inputreader.to.InputFileTo;
+import com.devonfw.cobigen.tsplugin.merger.constants.Constants;
+
+/**
+ *
+ */
+public class TypeScriptInputReader implements InputReader {
+
+    /** Logger instance. */
+    private static final Logger LOG = LoggerFactory.getLogger(TypeScriptInputReader.class);
+
+    /**
+     * Instance that handles all the operations performed to the external server, like initializing the
+     * connection and sending new requests
+     */
+    private ExternalProcessHandler request = ExternalProcessHandler
+        .getExternalProcessHandler(ExternalProcessConstants.HOST_NAME, ExternalProcessConstants.PORT);
+
+    /**
+     * Exception handler related to connectivity to the server
+     */
+    private ConnectionExceptionHandler connectionExc = new ConnectionExceptionHandler();
+
+    /**
+     * Creates a new {@link TypeScriptInputReader}
+     */
+    public TypeScriptInputReader() {
+
+        try {
+            // We first check if the server is already running
+            request.startConnection();
+            if (request.isNotConnected()) {
+                startServerConnection();
+            }
+        } catch (IOException e) {
+            // If it is not currently running, we need to execute it
+            LOG.info("Server is not currently running. Let's initialize it");
+            startServerConnection();
+        }
+
+    }
+
+    /**
+     * Deploys the server and tries to initialize a new connection between CobiGen and the server
+     */
+    private void startServerConnection() {
+        request.executingExe(Constants.EXE_NAME, this.getClass());
+        request.initializeConnection();
+    }
+
+    @Override
+    public boolean isValidInput(Object input) {
+        String fileContents = null;
+        String inputCharset = "UTF-8";
+
+        Path path;
+
+        if (input instanceof Path) {
+            path = (Path) input;
+        } else if (input instanceof File) {
+            path = ((File) input).toPath();
+        }
+
+        else {
+            try {
+                // Input corresponds to the parsed file
+                Map<String, Object> mapModel = createModel(input);
+                mapModel = (Map<String, Object>) mapModel.get("model");
+                path = Paths.get(mapModel.get("path").toString());
+                return isValidInput(path);
+            } catch (NullPointerException e) {
+                return false;
+            }
+
+        }
+
+        if (request.isNotConnected()) {
+            startServerConnection();
```

Comment:
Are you sure you need to start the server connection again? If you have the feeling that it's needed, it's okay for me. But it seems to me a bit redundant as in the constructor we already have called `startServerConnection();`